internal/hcsoci/create.go

```diff
@@ -236,8 +235,17 @@ func configureSandboxNetwork(ctx context.Context, coi *createOptionsInternal, r
 		// container but not a workload container in a sandbox that inherits
 		// the namespace.
 		if ct == oci.KubernetesContainerTypeNone || ct == oci.KubernetesContainerTypeSandbox {
-			if err := SetupNetworkNamespace(ctx, coi.HostingSystem, coi.actualNetworkNamespace); err != nil {
-				return err
+			if err := coi.HostingSystem.ConfigureNetworking(ctx, coi.actualNetworkNamespace); err != nil {
+				if err == uvm.ErrNoNetworkSetup {
+					// No network interface passed in, just set up locally.
+					coi.HostingSystem.NetworkSetup = uvm.NewInternalNetworkSetup(coi.HostingSystem)
+					err := coi.HostingSystem.ConfigureNetworking(ctx, coi.actualNetworkNamespace)
+					if err != nil {
+						return err
+					}
```

Comment:
do we want to use `CreateAndAssignNetworkSetup` here?

Reply:
Sure, makes more sense if we're using everywhere else